message-transport/pkg/transport/stubtransport/consumer.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018-Present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stubtransport
+
+import (
+	"github.com/projectriff/riff/message-transport/pkg/message"
+	"github.com/projectriff/riff/message-transport/pkg/transport"
+)
+
+// This stub is for use in tests.
```

Comment:
Can this live in a `_test.go` file?

Reply:
Unfortunately not. Renaming the file produces the warning:
```
go build [...]/stubtransport: no non-test Go files in [...]/stubtransport
```

Adding a `package.go` doesn't help. The users of `stubtransport` then fail to compile!

I think this is a limitation of Go - see [various](https://github.com/golang/go/issues/8279) [questions](http://grokbase.com/t/gg/golang-nuts/136t4f5een/go-nuts-importing-from-test-packages) users have raised.

Note that mocks don't conform to that naming convention, so I think it's ok that these stubs do too.

Reply:
That's because this is a different package. What about moving it to the original package?

Or are they needed by stuff and tests outside that package?

Reply:
I don't believe this is implementable. Please let me know if you find a solution.